app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -1557,18 +1559,40 @@ private void saveDraftAndFinish() {
                     if (accountList != null) {
                         resultList.addAll(accountList);
                     }
-                    return CollectionsKt.map(resultList, MentionTagAutoCompleteAdapter.AccountResult::new);
+                    return CollectionsKt.map(resultList, ComposeAutoCompleteAdapter.AccountResult::new);
                 case '#':
                     Response<SearchResults> response = mastodonApi.search(token, false).execute();
                     if (response.isSuccessful() && response.body() != null) {
                         return CollectionsKt.map(
                                 response.body().getHashtags(),
-                                MentionTagAutoCompleteAdapter.HashtagResult::new
+                                ComposeAutoCompleteAdapter.HashtagResult::new
                         );
                     } else {
                         Log.e(TAG, String.format("Autocomplete search for %s failed.", token));
                         return Collections.emptyList();
                     }
+                case ':':
+                    try {
+                        emojiListRetrievalLatch.await();
+                    } catch (InterruptedException e) {
+                        Log.e(TAG, String.format("Autocomplete search for %s was interrupted.", token));
+                        return Collections.emptyList();
+                    }
+                    if (emojiList != null) {
+                        final String incomplete = token.substring(1);
+
+                        List<Emoji> results = CollectionsKt.filter(
+                                emojiList,
+                                emoji -> emoji.getShortcode().startsWith(incomplete)
```

Comment:
Why only `startsWith`? Wouldn't it be better to return all emojis that contain the string, with emojis that start with it first?

Reply:
Wow, that's a great idea! The second list will be a lot longer than the first in many cases, so maybe it'd be a good idea to put a horizontal rule or something between them?

I should probably also make the matching case-insensitive.